src/qt/paymentserver.cpp

```diff
@@ -567,27 +567,18 @@ void PaymentServer::fetchPaymentACK(CWallet* wallet, SendCoinsRecipient recipien
     payment.add_transactions(transaction.data(), transaction.size());
 
     // Create a new refund address, or re-use:
-    QString account = tr("Refund from %1").arg(recipient.authenticatedMerchant);
-    std::string strAccount = account.toStdString();
-    std::set<CTxDestination> refundAddresses = wallet->GetAccountAddresses(strAccount);
-    if (!refundAddresses.empty()) {
-        CScript s = GetScriptForDestination(*refundAddresses.begin());
+    CPubKey newKey;
+    if (wallet->GetKeyFromPool(newKey)) {
+        CKeyID keyID = newKey.GetID();
+        std::string label = tr("Refund from %1").arg(recipient.authenticatedMerchant).toStdString();
+        wallet->SetAddressBook(keyID, label, "refund");
```

Comment:
(commit d597921101b060e7d9e9071f16bb385762a4206d) maybe add a new const string in the namespace `AddressBook::AddressBookPurpose` for this purpose "refund"

Reply:
I wouldn't care much about this instance for now, the paymentserver is not enabled (needs review and further discussion to know if we really want it or not). The "refund" purpose is not connected to the GUI neither.

Reply:
yeah, we should decide if we even want to re-implement this paymentserver or completely remove it.